### PR TITLE
Clippy fixes for: tedge

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Proposed changes

This is part of #825 - my patches to fix the clippy warnings in the `tedge` crate.

## Types of changes

- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge/src/cli/connect/command.rs [hunk 1]
```diff
@@ -74,9 +74,9 @@ impl Command for ConnectCommand {
             let br_config = self.bridge_config(&config)?;
             if self.check_if_bridge_exists(&br_config) {
                 return match self.check_connection(&config) {
-                    Ok(DeviceStatus::AlreadyExists) => {
-                        let cloud = br_config.cloud_name.clone();
-                        println!("Connection check to {} cloud is successful.\n", cloud);
+                    Ok(_) => {
+                        let cloud = br_config.cloud_name;
+                        println!("Connection check to {} cloud is successfull.\n", cloud);
```
Comment (makr11st) [inline crates/core/tedge/src/cli/connect/command.rs:79]: Previous spelling was correct :)
```suggestion
                        println!("Connection check to {} cloud is successful.\n", cloud);
```

Comment (matthiasbeyer) [reply crates/core/tedge/src/cli/connect/command.rs:79]: Whoopsie, that slipped. Thanks!

**Comment by matthiasbeyer** [comment]
Fixup commits will be squashed before merging of course.

File: crates/core/tedge/src/cli/connect/command.rs [hunk 2]
```diff
@@ -74,8 +74,8 @@ impl Command for ConnectCommand {
             let br_config = self.bridge_config(&config)?;
             if self.check_if_bridge_exists(&br_config) {
                 return match self.check_connection(&config) {
-                    Ok(DeviceStatus::AlreadyExists) => {
-                        let cloud = br_config.cloud_name.clone();
+                    Ok(_) => {
+                        let cloud = br_config.cloud_name;
```
Comment (makr11st) [inline crates/core/tedge/src/cli/connect/command.rs:78]: @rina23q I believe this check against a particular variant was done for a reason, could you have a look at this?

Comment (rina23q) [reply crates/core/tedge/src/cli/connect/command.rs:78]: Thanks for catching it! Yes, Pradeep and I worked on the fix of this condition by #861. 
We have a variant `Ok(DeviceStatus::Unknown)`, which we should not print "successful" in the output.
With the variant, the connection check has publish a message successfully, however, we didn't get the expected response by cloud for some reason. That's why we can't print "successful".

**Review by rina23q — CHANGES_REQUESTED** [review]
Thanks to @makr11st, there is regression against the change we made in #860 and #861.
Others look very good. After the Ok() condition gets fixed, I will approve.

**Comment by matthiasbeyer** [comment]
Please find my fixup commit and review whether you approve for this change.
If you approve, I will squash it.

**Review by makr11st — APPROVED** [review]

**Review by rina23q — APPROVED** [review]

**Pull request merged**